src/plugins/walletConnectToDapps/components/DappRegistryGrid.tsx

```diff
@@ -0,0 +1,96 @@
+import { SearchIcon } from '@chakra-ui/icons'
+import {
+  Box,
+  Heading,
+  Image,
+  Input,
+  InputGroup,
+  InputLeftElement,
+  Link,
+  SimpleGrid,
+  Stack,
+  Text as PlainText
+} from '@chakra-ui/react'
+import { Text } from 'components/Text'
+import { FC, useEffect, useMemo } from 'react'
+import { useForm } from 'react-hook-form'
+
+interface RegistryItem {
+  category: string
+  id: string
+  homepage: string
+  name: string
+  image: string
+}
+
+const registry: RegistryItem[] = require('../registry.json')
+
+const pageSize = 20
+
+export const DappRegistryGrid: FC = () => {
+  const { register, watch, setValue } = useForm<{ search: string; page: number }>({
+    mode: 'onChange',
+    defaultValues: {
+      search: '',
+      page: 0,
+    },
+  })
+
+  const search = watch('search')
+  const page = watch('page')
```

Comment:
Use `useWatch()` instead of watch: the former is react-friendly, the latter isn't and will require you to unsubscribe to watched inputs
https://react-hook-form.com/api/useform/watch